### ISSUE TITLE
Files deleting while editing; Dropbox folder as destination

### ISSUE THREAD

**Issue by infews** [ttscoff/nv#514]
While editing a file, that file gets deleted in the background. Yikes!

I'm running 2.2.8 (128), MacOS Sierra 10.12.6, and the latest Dropbox client. MacBook Air

The only thing I think could be triggering it - and this sounds CRAZY - is an accidental touchpad touch of my right hand while typing.

I've done some searching and not found anything useful. So I figured I would drop this here.

**Comment by thadk** [comment]
Here is the error Dropbox is giving when the file almost gets deleted on my machine.
![image](https://user-images.githubusercontent.com/283343/41136399-1806d428-6aa4-11e8-8dd7-b7ea7da9ea86.png)

**Comment by thadk** [comment]
It was also noted in the last comment of this thread in January: https://github.com/ttscoff/nv/issues/129

**Comment by jacobweber** [comment]
Just noticed this during the last week or so, including the alerts from Dropbox. Using Mac OS 10.13.5, Dropbox 52.4.58, NVAlt 2.2.8.

**Comment by PaulRein** [comment]
I also started noticing the same messages as thadk showed, in the last couple of days. It is extremely annoying since it pops up every few minutes which steals focus from nvALT so my writing gets interrupted. It is very annoying to constantly try to find where the cursor was when I am writing/editing in the middle of a paragraph when Dropbox interrupts. It does not seem to help to check the "Don't ask me this again"-box either, the message is back a couple of minutes later anyway.

It has only been the last two days (since 2018-06-28) for me but it seems to increase in frequency. The first day it was only a few times in the day (sparse enough that I didn't make the connection at first) but now it is every few minutes, making nvALT all but useless right now. (And it is only with nvALT it happens, Dropbox has _not_ notified me while doing anything else.)

It is really frustrating since the "hotkey-nvALT, dash down a quick note, hotkey nvALT-away" workflow is so ingrained in me that it normally does not interrupt my flow of what I am doing but now it comes to a screeching halt when Dropbox interrupts and I need to check how much of the note made it.  

Using Mac OS 10.13.5, Dropbox 52.4.60, nvALT 2.2.8 (128). I only have a single machine running nvALT right now, so it is not conflicting instances like in the referenced old thread.

**Comment by thadk** [comment]
Please also report which Storage format you're using. I'm on "Plain text files".
Also maybe note if you have any Conflicted Copies of the `Notes & Settings` file. That seemed to appear around the times the problems started:
![image](https://user-images.githubusercontent.com/283343/42127523-88009222-7c57-11e8-850b-6e248d555428.png)

**Comment by jacobweber** [comment]
I'm using plain text format. I did have a bunch of those conflicted copies.

**Comment by PaulRein** [comment]
Also using plain text format. No conflicting copies of Notes & Settings.

**Comment by jacobweber** [comment]
Just got it again, and this time there's no conflicted Notes & Settings.

**Comment by Shl1ck** [comment]
This problem never happened before with me, but matches the kind of stories the other people above are saying. The additonal fear issue is that the options given to Cancel, Delete everywhere, and Don't ask me this again, is that I might not be "asked again" and find files disappearing.
I presume this has to do with the current  update 2.2.8. 

I have deleted "Notes & Settings 2" an older file than "Notes & Settings" and will see how that affects the problem., if at all.

**Comment by khorton** [comment]
I've been afflicted with this problem recently as well, with nvALT 2.2.8, macOS 10.13.5 and DropBox v53.4.67.  I stopped nvALT, deleted all Conflicted Copies of Notes & Settings, and restarted.  But the problem continues every few minutes.

**Comment by thadk** [comment]
Anyone know what happens when we erase the main "Notes & Settings" file and have it make a new one?

**Comment by mhairston** [comment]
This has been happening to me as well, with DropBox as the storage method. I have found that:

* It is DropBox that is generating those dialog boxes. Try going to the Finder and deleting a file that is synced by DropBox, and you'll get the same dialog.
* When it happens while editing in nvAlt (and the file mentioned is the file you're editing), try this: Copy all the contents of the file into your clipboard for safety, then answer the dialog "Delete everywhere". The file will still be there.

My theory is that somehow, the way nvAlt or the file system is auto-saving the file, it looks to DropBox as if the file is being deleted. And my guess is that this confirm-delete behavior was introduced in DropBox fairly recently.

This is definitely bad behavior, as it affects your confidence in the safety of your data, but has anyone actually lost any files?

**Comment by thadk** [comment]
Maybe we could escalate this to Dropbox then, with their more ample troubleshooting resources?

**Comment by matthewsinclair** [comment]
FWIW, I can confirm the exact experience of @mhairston above.

**Comment by hay** [comment]
Yup, same problem here. Using NvAlt 2.2.8, Dropbox v53.4.67 and MacOS 10.13.5. 

For now i think i'll just switch off the syncing for my notes :(

**Comment by thadk** [comment]
Dropbox responded that they don't want to do anything even though their new version caused the problem. Because the warnings are false alarms, the best thing to do is to hide the warnings until someone can figure out what is triggering this in nvalt code and incorporate the fix into a release.

**Comment by infews** [comment]
Just coming back after a bit - my notifications got lost.

I can confirm that I'm also using text files to a folder on Dropbox. Mac OS 10.12.6 and another client on 10.13.something. Dropbox 53.4.67.

Interestingly, since I logged this issue, Dropbox has started nagging with these dialog boxes. I find it an (annoying) workaround to use this dialog to prevent the deletion. So I'm happy-ish at the moment.

But I agree that this is probably something that nvALT is doing that other apps are not.

**Comment by infews** [comment]
Has anyone noticed the delete requests coming based on typing speed? As in, after several key presses in short succession triggering the Dropbox delete dialog?

**Comment by matthewsinclair** [comment]
Yes. Mine seems to be arrow key related.

**Comment by infews** [comment]
I'm seeing arrow and possibly non-letter keys as well.

**Comment by seranki** [comment]
Sharing my config because I'm not having these problems:
MacBook Pro (late 2013) ; nvALT version 2.2.8 (128) ; macOS 10.13.6

Dropbox 54.4.90
Dropbox Preferences > Notifications:
- **all turned off except "Files and folders shared with me"**

nvALT Preferences:
- notes in a Dropbox folder 
- Sync: none
- Storage: as Plain Text Files
- Security: No Note Encryption ; No Secure Text Entry

For what it's worth, the 1Writer app on iOS works perfectly with nvALT ; reads and writes in both directions
Hope this helps

**Comment by thadk** [comment]
Dropbox should not have added a feature which interferes with basic app functionality unless it can filter apps that do not support the feature.

**Comment by thadk** [comment]
This is now fixed for me. I think we can close it!
I did untick this "New files synced" setting and that may have helped:
![image](https://user-images.githubusercontent.com/283343/50550310-3f3c5380-0c3c-11e9-97f3-e77c0fa71257.png)

**Comment by afenc** [comment]
I've tried most of the above mentioned remedies, none of them works. 
My hypothesis is the strange interaction between Notes & Settings file and dropbox sync mechanism.
Here is my trial:
1. backup the note folder
2. delete all note files and "Notes & Settings" file in that folder
3. open the nvALT, which should be empty now
4. import the notes to nvALT, which will generate a new "Notes & Settings"

This rebuild database works for me. The major drawback is that all titles of note have gone.

**Comment by thadk** [comment]
@afenc I had to do something like this is the past, if you change to plaintext and not a database, I think nvalt would have written out everything as individual files with titles as filenames and the correct dates to a single folder. Then you can use the preserve timestamps on the files using `cp` to make a backup copy or choose a different folder.
